### PR TITLE
Change currency based on location.

### PR DESCRIPTION
**Pull request by akatsoulas**

### CODE REVIEW

**Comment by akatsoulas** [comment]
@comzeradd or @johngian r?

File: marketpulse/main/static/js/price_new.js [hunk 1]
```diff
@@ -23,6 +24,11 @@ $(document).ready(function() {
         }
     }
 
+    function changeCurrency () {
+        var currency = $('.formset-currency:first').val();
+        $('.formset-currency:not(:first)').val(currency);
```
Comment (johngian) [inline marketpulse/main/static/js/price_new.js:29]: Instead of using `:not(:first)` you could use `nextAll`.

File: marketpulse/main/views.py [hunk 2]
```diff
@@ -23,14 +26,26 @@ def edit_fxosprice(request, username='', id=None):
...
         lat = request.GET.get('latitude')
         lng = request.GET.get('longitude')
 
-        if lat and lng:
-            location_data = reverse_geocode(lat, lng)
-            country_code = location_data.get('country')
-        return JsonResponse({'country': country_code})
+        data = {'country': None,
+                'currency': None}
+        if not lat or not lng:
```
Comment (johngian) [inline marketpulse/main/views.py]: You could avoid this and return `JsonResponse` directly in the end.

File: marketpulse/main/views.py [hunk 3]
```diff
@@ -23,14 +26,26 @@ def edit_fxosprice(request, username='', id=None):
...
+            return JsonResponse(data)
+
+        location_data = reverse_geocode(lat, lng)
+        country_code = location_data.get('country')
+        if country_code:
+            data['country'] = location_data.get('country')
```
Comment (johngian) [inline marketpulse/main/views.py]: Please use `country_code` directly.

**Comment by johngian** [comment]
LGTM r+wc

**Comment by comzeradd** [comment]
We could also change the currency based on the country field for online shops, but this can also be a separate PR.

**Comment by akatsoulas** [comment]
The country field for online shops is also populated from the reverse geo-location. If the reverse lookup fails, then the user will have to manually set the country if the shop is online

File: marketpulse/main/static/js/price_new.js (hunk 1 above)
Comment (akatsoulas) [inline marketpulse/main/static/js/price_new.js:29]: Unfortunately, it does not work with `nextAll`. Leaving this as is

**Pull request merged**